athena-federation-integ-test/README.md

```diff
@@ -88,7 +88,35 @@ Provide implementation for the following 4 abstract methods in the test class:
      * access to multiple connector-specific AWS services (e.g. DynamoDB, Elasticsearch etc...)
      * @return A policy document object.
      */
-    protected abstract PolicyDocument getConnectorAccessPolicy();
+    protected abstract Optional<PolicyDocument> getConnectorAccessPolicy();
+```
+
+### VPC Configuration
+
+The Integration-Test module can be adapted to test connectors that utilize a VPC configuration to connect to the data
+source. The framework can detect the default VPC configuration set in the specified AWS account and use it to configure
+the Lambda function. In order for the latter to be able to connect to the data source, however, the same VPC configuration
+must be set when provisioning the DB instance.
+
+To use a VPC configuration, the following method must be overridden in your integration test class to `return true`
+(default is false):
+
+```java
+    /**
+     * Must be overridden in the extending class when the connector supports a VPC configuration.
+     * @return false (default)
+     */
+    protected boolean connectorSupportsVpcConfig()
+```
+Access to the VPC configuration (VPC Id, Security Group Id, Subnet Ids, and Availability Zones) can be obtained via the
+Stack object:
+
+```java
+    @Override
+    protected void setUpStackData(final Stack stack)
+    {
+        ConnectorVpcAttributes vpcAttributes = ((ConnectorWithVpcStack) stack).getConnectorVpcAttributes();
```

Comment:
do you really need the boolean or can you just make getCOnnectorVpcAttributes() return an Optional?

Reply:
Currently the fwk uses the boolean in the stack factory to determine what kind of stack to generate (with/without VPC). This will need to be reworked when switching to a config file. The values from the config file, thus, can drive the creation of the stack rather than a boolean return value.